app/javascript/controllers/chart_controller.js

```diff
@@ -19,8 +20,38 @@ export default class extends Controller {
   }
 
   showChart() {
+    var datesModify
+    if(this.aggValue === "day") {
+      datesModify = this.datesValue.map(function(e) { 
+        var d = new Date(e);
+        const options = { weekday: 'short', year: '2-digit', month: 'short', day: '2-digit' };
+        d = d.toLocaleDateString('en-GB', options);
+        return d;
+      })
+    } else if(this.aggValue === "week") {
+      datesModify = this.datesValue.map(function(e) { 
+        var d = new Date(e);
+        const options = { year: 'numeric', month: 'short', day: '2-digit' };
+        d = d.toLocaleDateString('en-GB', options);
+        return d;
+      })
+    } else if(this.aggValue === "month") {
+      datesModify = this.datesValue.map(function(e) { 
+        var d = new Date(e);
+        const options = { year: 'numeric', month: 'short' };
+        d = d.toLocaleDateString('en-GB', options);
+        return d;
+      })
+    } else if(this.aggValue === "year") {
+      datesModify = this.datesValue.map(function(e) {
+        var d = new Date(e);
+        const options = { year: 'numeric'};
+        d = d.toLocaleDateString('en-GB', options);
+        return d;
+      })
+    };
```

Comment:
I would change a few of things to improve the code quality:
- Move the logic to a separate function instead of being inside the `showChart()` method
- There's a lot of repetition - Can you think of ways to refactor so that you don't repeat yourself (DRY = "don't repeat yourself" is an important concept in programming)
- Instead of using `if...else`, can you think of a better-suited condition statement?
- You're using `const`, that's great! In modern JS, the use of `var` is discouraged. What can you use instead?
- Try to be consistent with the quotes double (`"`) vs  single (`'`) - see what's used in this codebase and stick to that
- Let's go with `en-US` instead of `en-GB`